src/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -3144,12 +3145,12 @@ public static void townAdd(CommandSender sender, Town specifiedTown, String[] na
 	}
 
 	// wrapper function for non friend setting of perms
-	public static void setTownBlockOwnerPermissions(Player player, TownBlockOwner townBlockOwner, String[] split) {
+	public static void setTownBlockOwnerPermissions(Player player, Permissible townBlockOwner, String[] split) {

```

Comment:
I don't really understand why the parameter type was changed when it seems like you are checking if they are a `TownBlockOwner` anyway.

Reply:
I don't see why this is changed. I find it more readable. I don't prefer what it has been changed to either.

Reply:
Because as stated in the description I wanted to decouple `TownBlockOwner` from `Permissible` as we may want to adopt `TownBlock` owner but not `Permissible`. Using the generic parameter `<T extends TownBlockOwner & Permissible>` helps enforce type safety by preventing `instanceOf` calls followed by casting.